src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -705,36 +706,34 @@ search_and_program_xclbin(const std::shared_ptr<xrt_core::device>& dev, boost::p
   std::string xclbin = ptTest.get<std::string>("xclbin", "");
 
   //if no xclbin is loaded, locate the default xclbin
-  if (uuid_is_null(uuid) && !xclbin.empty()) {
-    //check if a 2RP platform
-    std::vector<std::string> logic_uuid;
-    try{
-      logic_uuid = xrt_core::device_query<xrt_core::query::logic_uuids>(dev);
-    } catch(...) { }
-
-    std::string xclbinPath;
-    auto xclbin_location = ptTest.get<std::string>("xclbin_directory", "");
-    if(!xclbin_location.empty()) {
-      xclbinPath = xclbin_location + xclbin;
-    }
-    else if(!logic_uuid.empty()) {
-      xclbinPath = searchSSV2Xclbin(logic_uuid.front(), xclbin, ptTest);
-    } else {
-      auto vendor = xrt_core::device_query<xrt_core::query::pcie_vendor>(dev);
-      auto name = xrt_core::device_query<xrt_core::query::rom_vbnv>(dev);
-      xclbinPath = searchLegacyXclbin(vendor, name, xclbin, ptTest);
-    }
+  //check if a 2RP platform
+  std::vector<std::string> logic_uuid;
+  try{
```

Comment:
```suggestion
  try {
```
Add a comment on why we need this.

Reply:
The comment above the try statement is correct. This code block is needed to validate if the device is a 2RP system.